### PR TITLE
mdoc: Multiple performance fixes

### PR DESCRIPTION
**Pull request by rolfbjarne**
Multiple performance fixes that makes processing Xamarin.iOS.dll go from
taking 25 minutes to 26 seconds.

### CODE REVIEW

**Comment by rolfbjarne** [comment]
There's an identical PR for mono: https://github.com/mono/mono/pull/4729

**Comment by joelmartinez** [comment]
awesome! I'll run tests and merge this on Monday, probably :)

File: mdoc/Mono.Documentation/monodocer.cs [hunk 1]
```diff
@@ -4265,9 +4315,9 @@ public DocumentationMember (XmlNode node)
 	void DiscernTypeParameters ()
 	{
 		// see if we can discern the param list from the name
-		if (MemberName.Contains ("<") && MemberName.EndsWith (">")) {
-			var starti = MemberName.IndexOf ("<") + 1;
-			var endi = MemberName.LastIndexOf (">");
+		if (MemberName.Contains ("<") && MemberName.EndsWith (">", StringComparison.Ordinal)) {
+			var starti = MemberName.IndexOf ("<", StringComparison.Ordinal) + 1;
```
Comment (marek-safar) [inline mdoc/Mono.Documentation/monodocer.cs]: Both should use `char` overload

File: mdoc/Mono.Documentation/monodocer.cs [hunk 2]
```diff
@@ -4265,9 +4315,9 @@ public DocumentationMember (XmlNode node)
 	void DiscernTypeParameters ()
 	{
 		// see if we can discern the param list from the name
-		if (MemberName.Contains ("<") && MemberName.EndsWith (">")) {
-			var starti = MemberName.IndexOf ("<") + 1;
-			var endi = MemberName.LastIndexOf (">");
+		if (MemberName.Contains ("<") && MemberName.EndsWith (">", StringComparison.Ordinal)) {
```
Comment (marek-safar) [inline mdoc/Mono.Documentation/monodocer.cs]: There is duplicate search for `<` just one line bellow

**Comment by joelmartinez** [comment]
@rolfbjarne is this ready to review?

File: mdoc/Mono.Documentation/monodocer.cs (hunk 1 above)
Comment (rolfbjarne) [inline mdoc/Mono.Documentation/monodocer.cs]: Fixed: a3d677d371808728a7d0c4e31b01a384f4c96ec2

File: mdoc/Mono.Documentation/monodocer.cs (hunk 2 above)
Comment (rolfbjarne) [inline mdoc/Mono.Documentation/monodocer.cs]: Fixed: 4473132fcce71bbdb593e325f6febe252f3713fa

**Comment by rolfbjarne** [comment]
@joelmartinez yes, please review :smile:

**Comment by joelmartinez** [comment]
Hi Rolf ... apologies for the delay in reviewing this PR, but finally getting to it now. I pulled this branch, and rebased it locally, in order to verify the tests ... unfortunately, the test suite is not passing :( 

Please rebase, and then make whatever changes necessary to get the tests passing:

```bash
make prepare all check -B
```

I briefly looked through the PR, and offhand I can't see why it's changing some of the output of the tests that I'm seeing.

**Comment by joelmartinez** [comment]
build

**Comment by rolfbjarne** [comment]
Closing, this has bitrotted.

**Pull request closed without merging**